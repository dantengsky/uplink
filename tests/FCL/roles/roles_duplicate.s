global account admin = u'6pxGdGG6nQP3VoCW7HoGkCGDNCiCEWP3P5jHtrvgphBc';
global account bob = u'CGDNCiCEWP3P5jHtrvgphBc6pxGdGG6nQP3VoCW7HoGk';

transition initial -> terminal;

@initial { u'6pxGdGG6nQP3VoCW7HoGkCGDNCiCEWP3P5jHtrvgphBc' // ok
         , u'6pxGdGG6nQP3VoCW7HoGkCGDNCiCEWP3P5jHtrvgphBc' // bad
         , deployer() // ok
         , deployer() // bad
         , admin // ok
         , admin // bad
         , admin // bad
         , bob // ok
         }
init() {
 
  terminate("Bye.");
}