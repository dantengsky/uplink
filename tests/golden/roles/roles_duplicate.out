Duplicate variables found:
 > Duplicate role in method access control: u'6pxGdGG6nQP3VoCW7HoGkCGDNCiCEWP3P5jHtrvgphBc'
 > Duplicate role in method access control: deployer()
 > Duplicate role in method access control: admin